profile/src/main.rs

```diff
@@ -0,0 +1,152 @@
+use std::{env,fs::File};
+use polars::prelude::{LazyFrame,LazyGroupBy};
+use polars::prelude::*;
+use temp_name_lib::type_def::{CLIGHT, N_FLUX_POINTS};
+use profile::*;
+use std::sync::Arc;
+//use pulstar::utils::PulstarConfig;
+
+fn main() {
+
+   let env_args:Vec<String> = env::args().collect(); 
+
+   if env_args.len() < 2usize {
+    println!("Not enough arguments.");
+    panic!("USAGE: profile -- <profile_input.txt>");
+   }
+
+   let profile_config=config{
+    lambda_0:412.0,
+    lambda_f:413.0,
+    delta_lbd:0.01,
+    v_max:1.0e2,
+    n_phases:10,
+   };
+
+
+   let capacity = ((profile_config.lambda_f - profile_config.lambda_0)
+        /profile_config.delta_lbd).floor() as usize + 1usize ;
+
+    if capacity >= N_FLUX_POINTS as usize {panic!("Error, too many flux points requested.")}
+
+    let mut wavelength:Vec<f64> = Vec::with_capacity(capacity);
+    
+    wavelength.push(profile_config.lambda_0);
+
+    for i in 0..=capacity {//<- inclussive loop so wavelength[capacity]==λ_f.
+            wavelength.push(profile_config.lambda_0 
+                + profile_config.delta_lbd * i as f64 );
+    }
+
+    let lambda_lower = wavelength[0] * (1.0 - profile_config.v_max/CLIGHT*1.0e3);
+    let lambda_upper= wavelength.last().unwrap() * (1.0 + profile_config.v_max/CLIGHT*10.0e3);
+
+
+    let mut flux = vec![0.0;capacity];
+    let mut cont = vec![0.0;capacity];
+    //open parquet file with pulsations
+    //open txt file as csv !heel moeilijk
+
+    //group by timepoints
+
+    //Loop over the phases
+//    for i in 0..=profile_config.n_phases{
+//        let mut flux=vec![0.0;capacity];
+//       let mut continuum = vec![0.0;capacity];
+        
+
+
+        //----------------------------------------
+        // Start the loop over all points of the visible surface
+        // Compute the local line profile in each point. 
+        // Add the appropriate contribution to each point of the global line profile.
+        
+        // lambda_rest = wavelength[j]/(1.0_Vtotal/CLIGHT * 1.0e3)
+
+
+        //write the profile lines!<- gold!
+//    }
+    let path= String::from("pulstar_output.parquet");
+    let lf = LazyFrame::scan_parquet(path, Default::default()).unwrap();
+    
+    //get vector of time_points
+    let tf = lf.clone().select([col("time").unique(),]).collect().unwrap();
+    let extract_time_series = tf.column("time").unwrap();
+    let time_points:Vec<f64> = extract_time_series.f64().unwrap().into_iter().flatten().collect();
+
+    for phases in time_points.iter() {
+        //get vector of theta points
+        let expr = col("time").eq(lit(*phases));
```

Comment:
`expr` is not a great variable name. How about `current_phase`. Then an expression like `lf.filter(current_phase)` becomes much more readable.